include/circt-c/Dialect/FIRRTL.h

```diff
@@ -105,28 +105,73 @@ MLIR_DECLARE_CAPI_DIALECT_REGISTRATION(FIRRTL, firrtl);
 // Type API.
 //===----------------------------------------------------------------------===//
 
+/// Returns `true` if this is a const type whose value is guaranteed to be
+/// unchanging at circuit execution time.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsConst(MlirType type);
+
+/// Returns a const or non-const version of this type.
+MLIR_CAPI_EXPORTED MlirType firrtlTypeGetConstType(MlirType type, bool isConst);
+
+/// Gets the bit width for this type, returns -1 if unknown.
+///
+/// It recursively compute the bit width of aggregate types. For bundle and
+/// vectors, recursively get the width of each field element and return the
+/// total bit width of the aggregate type. This returns -1, if any of the bundle
```

Comment:
> This returns -1, if any of the bundle fields is a flip type

why

Reply:
This is the design of https://github.com/llvm/circt/blob/cd02a733b22cd8a8f45b860c6caebd5621347b6e/include/circt/Dialect/FIRRTL/FIRRTLTypes.h#L376-L383

But I forgot to present the `ignoreFlip` parameter in C-API, I should add it as well.

Reply:
Updated~